pkg/pki/x509/x509.go

```diff
@@ -176,9 +178,11 @@ func (k PublicKey) CanonicalValue() ([]byte, error) {
 func (k PublicKey) EmailAddresses() []string {
 	var names []string
 	if k.cert != nil {
-		for _, name := range k.cert.c.Subject.Names {
-			if name.Type.Equal(EmailAddressOID) {
-				names = append(names, strings.ToLower(name.Value.(string)))
+		for _, name := range k.cert.c.EmailAddresses {
+			validate := validator.New()
+			errs := validate.Var(name, "required,email")
+			if errs == nil {
+				names = append(names, strings.ToLower(name))
```

Comment:
maybe toss in a debug log or something here if it doesn't validate?

Reply:
would the server care? these fields aren't validated by cert issuers so anything goes with these fields. (e.g. you can toss an email for the Organization field)
but this is just so we are only indexing actual emails.

Reply:
Not really, just more to help us with debugging in case the validation didn't work or something.